packages/swingset-liveslots/test/test-cache.js

```diff
@@ -26,12 +26,9 @@ test('cache', t => {
 
   // new reads pass through immediately to backing store
   t.is(c.get('key0'), 'value0');
+  t.deepEqual(log.splice(0), [['read', 'key0']]);
```

Comment:
`splice` is a confusing operation that I have to look up the docs for each time I use it.  I'm wondering if it would be clearer to have a helper function that consumes the head of the log and returns it (which of course would be written with `splice`, but only in one place).  Seems like we have something like that in other tests.

Reply:
Heh, see https://github.com/Agoric/agoric-sdk/pull/7138#discussion_r1154707811
> I like `splice` for supporting the ability to make assertions about a batch of entries rather than one at a time, and really this should be an external test helper where the implementation is explained inline but never really considered at call sites because it Just Works as expected (above suggestion updated with that in mind).
> 
> ```js
> /**
>  * Clear the contents of a log array and assert that they match expectations.
>  *
>  * @param {import('ava').ExecutionContext} t
>  * @param {unknown[]} log
>  * @param {unknown[]} expected
>  * @returns {void}
>  */
> const clearAndAssertLog = (t, log, expected) => {
>   // `log.splice(0)` removes elements starting at index 0 (i.e., every element)
>   // and returns them in an array.
>   const entries = log.splice(0);
>   t.deepEqual(entries, expected);
> }
> ```

Reply:
ok, two votes for a helper, I'll start using one in the future